src/components/Dropdown.js

```diff
@@ -80,29 +85,60 @@ export default class Dropdown extends React.Component {
     }
   }
 
+  handleKeys = event => {
+    if (this.refs.list) this.refs.list.handleKeys(event)
+  }
+
+  chooseChild = choice => {
+    const link = findChildLink(findDOMNode(this.refs[choice.ref]))
+    fireEvent(link, 'click')
+  }
+
   render () {
     const {
-      toggleChildren, children, alignRight, backdrop, triangle, openOnHover, id
+      toggleChildren, children, alignRight, backdrop, triangle, openOnHover, id,
+      keyControlled
     } = this.props
     const { hoverOpened } = this.state
     const { isMobile } = this.context
     const active = this.state.active && !isEmpty(children)
     const className = cx('dropdown', this.props.className,
       {active, 'has-triangle': triangle})
+    const ulProps = {
+      className: cx('dropdown-menu', {'dropdown-menu-right': alignRight}),
+      style: mobileMenuStyle(isMobile && active, this.refs.parent),
+      onClick: () => this.toggle(),
+      onMouseLeave: () => hoverOpened && this.toggle()
+    }
+
+    // adding refs to the children so that we can get the node in chooseChild above
+    const childrenWithRefs = React.Children.map(children,
+      (element, idx) => {
+        return element && element.props
+          ? React.cloneElement(element, {ref: idx})
+          : element
+      })
 
-    return <div id={id} className={className} ref='parent'>
+    const items = triangle
+      ? [<li className='triangle'
```

Comment:
Missed this. Right now we don't have a dropdown that is keyControlled and has triangle, but I'll fix it for future.